File: codewars/6kyu/checking-groups/dduck-checking-groups.js
// Checking Groups
// https://www.codewars.com/kata/54b80308488cb6cd31000161/

const regexp = /\(\)|{}|\[\]/gi;

const groupCheck = (s) => {
  if (s.match(regexp) && s.match(regexp).length) {
    return groupCheck(s.replace(regexp, ''));
  }
  return s.length === 0;
};

console.log(groupCheck("()"), "That group was matched correctly");
console.log(groupCheck("{(})"), "That group was matched incorrectly");
console.log(groupCheck("[])"), "That group was left open or prematurely closed");
console.log(groupCheck("()[]{}"), "That group was matched correctly");

File: codewars/6kyu/checking-groups/dduck-checking-groups.py
# Checking Groups
# https://www.codewars.com/kata/54b80308488cb6cd31000161/

from unittest import TestCase

import re

regexp = r"\(\)|{}|\[\]"


def group_check(s):
    if re.search(regexp, s):
        return group_check(re.sub(regexp, '', s))
    return len(s) == 0


TestCase().assertEqual(group_check("()"), True)
TestCase().assertEqual(group_check("({"), False)
TestCase().assertEqual(group_check("()"), True)
TestCase().assertEqual(group_check("{(})"), False)
TestCase().assertEqual(group_check("[])"), False)
TestCase().assertEqual(group_check("()[]{}"), True)
TestCase().assertEqual(group_check("[[]()]"), True)
